### PR TITLE
Test: cts-exec: Handle --list-tests and --help earlier in main()

### PR DESCRIPTION
**Pull request by waltdisgrace**

### CODE REVIEW

File: cts/cts-exec.in [hunk 1]
```diff
@@ -1225,6 +1225,9 @@ def main(argv):
     opts = TestOptions()
     opts.build_options(argv)
 
+    if opts.options['show-usage']:
+        tests.show_usage()
```
Comment (kgaillot) [inline cts/cts-exec.in:1229]: tests hasn't been created yet at this point. It is a good idea to do it before creating the temporary directory though, so maybe show_usage() should be taken out of the object and just be a regular function

we also need to exit after calling show_usage() (and after print_list() as well)

**Review by kgaillot — CHANGES_REQUESTED** [review]
It's always a good idea to test your changes :)

File: cts/cts-exec.in (hunk 1 above)
Comment (waltdisgrace) [reply cts/cts-exec.in]: Oops, I had accidentally tested the wrong code before I pushed... The next commit should fix it:

```
======= FINAL RESULTS ==========

--- FAILURE RESULTS:
    None

--- TOTALS
    Pass:53
    Fail:0
```

File: cts/cts-exec.in [hunk 2]
```diff
@@ -1226,7 +1226,8 @@ def main(argv):
     opts.build_options(argv)
 
     if opts.options['show-usage']:
-        tests.show_usage()
+       show_usage()
+       sys.exit(CrmExit.OK)
```
Comment (kgaillot) [inline cts/cts-exec.in:1230]: need four-space indent

**Review by kgaillot — COMMENTED** [review]
Looks great -- squash and it's ready to merge

**Review by kgaillot — APPROVED** [review]

**Pull request merged**